action-scripts/brownie/scripts/script_utils.py

```diff
@@ -432,25 +432,28 @@ def prettify_contract_inputs_values(chain: str, contracts_inputs_values: dict) -
     outputs = defaultdict(list)
     values = None
     for key, valuedata in contracts_inputs_values.items():
-        print(valuedata)
-        if isinstance(valuedata, list):
-            values = valuedata
-        elif isinstance(valuedata, str):
-            values = valuedata.strip("[ ]f").replace(" ", "").split(",")
-        if not isinstance(valuedata, list):
-            print(f"Warning f{values} is still not a list of values, putting what we have in single member list")
-            values = [valuedata]
+        values = parse_txbuilder_list_string(valuedata)
         for value in values:
+            ## Reverse resolve addresses
             if web3.isAddress(value):
                 outputs[key].append(
                     f"{value} ({addr.reversebook.get(web3.toChecksumAddress(value), 'N/A')}) "
                 )
-            elif "role" in key or "Role" in key:
+
+            ## Reverse resolve authorizor roles
+            elif "role" in key.lower():
                 outputs[key].append(
                     f"{value} ({perm.paths_by_action_id.get(value, 'N/A')}) "
                 )
+            ## If it looks like an amount provide 1e18 denomination (hard to understand what decimals will so make it clear.
+            elif "value" in key.lower() or "amount" in key.lower():
+                try:
+                    outputs[key].append(f"{value} / 1e18 = {int(value)/1e18}")
```

Comment:
are you sure we can always assume 18 decimals here?

Reply:
@gosuto-inzasheru No we can't...  The problem is, in a generalized report there is no way to know what decimals to assume.  That's why it clearly states over 1/e18 in the string so that someone can know what is going on and make a determination.  We could leave this out if you don't think it helps, but I often want for it.  Any thoughts?  Also it's not working/ trying to figure out this logic better if you have ideas there :)

In other news, we are running on 0.9.0 which is the new ecosystem-balances release/that looks good after some tweaks.

Reply:
Ok so I thought about this more.  When we know decimals or can know it I am using it.  When not will show both 18 and 6 decimals to make it clear that it is unclear that the actual answer is, but still be helpful.